ui/src/app/pages/admin/admin-user.tsx

```diff
@@ -136,32 +145,134 @@ const AdminUser = withUrlParams()(class extends React.Component<Props, State> {
     const {gsuiteDomain} = serverConfigStore.getValue();
     try {
       const profile = await profileApi().getUserByUsername(this.props.urlParams.usernameWithoutGsuiteDomain + '@' + gsuiteDomain);
-      this.setState({profile: profile});
+      this.setState({oldProfile: profile, updatedProfile: profile});
     } catch (error) {
       this.setState({profileLoadingError: 'Could not find user - please check spelling of username and try again'});
     }
   }
 
   async getInstitutions() {
     try {
-      const institutionsResponse = await institutionApi().getInstitutions();
-      const options = fp.map(
-        institution => {
-          return {
-            'label': institution.displayName ? institution.displayName : institution.shortName,
-            'value': {displayName: institution.displayName, shortName: institution.shortName}
-          };
-        },
-        institutionsResponse.institutions
-      );
-      this.setState({verifiedInstitutionOptions: options});
+      const institutionsResponse = await institutionApi().getPublicInstitutionDetails();
+      const institutions = institutionsResponse.institutions;
```

Comment:
It is good to think about the time complexity. My concern is around the code complexity - adding additional state and code to manage that state. My personal preference is for simpler code that is easier to maintain over time, even if that means in some cases it is not the most optimized. Part of my rationale for that is to prevent premature optimization and instead opt for simpler code. I feel removing the Map, reduce and additional state is worth the tradeoff. I do think your concern is valid, but if it does not cause a noticeable performance issue I would prefer managing less state in the component and if a performance issue does arise in the future we could address it.

Also, looking at this a bit more closely, it looks like the `find` would only occur on a selection change (it should not occur on a re-render - unless I am missing something). I ran a simple performance check in the browser comparing the find against the initial state setup with reduce (10,000 items). The reduce takes 1.8 - 3ms, whereas the find, in the worst case, takes about 0.6 - 0.8ms. You could run the find ~2-5 times in the amount of time it takes to run the reduce (which will occur on each mount). I can give you the code I ran if you are interested.

To me that makes it unclear which would be more performant in the real world - but again I would prefer to not avoid thinking too much about the optimization unless a performance problem became evident.

All that said - if it is too time consuming to make the change right now, I am ok with leaving it as is. I just wanted to better explain my position.

Reply:
Thank you for running a test! Because of email validation, the find will end up running on all renders. If it's a matter of milliseconds even for ten thousand things then I'll do the simpler code. This is what I get for assuming a new JS framework works like an ancient JS framework 🙂

Reply:
Thanks for doing that!